llvm/lib/Transforms/Vectorize/VPlanAnalysis.h

```diff
@@ -47,7 +49,8 @@ class VPTypeAnalysis {
   Type *inferScalarTypeForRecipe(const VPReplicateRecipe *R);
 
 public:
-  VPTypeAnalysis(LLVMContext &Ctx) : Ctx(Ctx) {}
+  VPTypeAnalysis(Type *CanonicalIVTy, LLVMContext &Ctx)
+      : CanonicalIVTy(CanonicalIVTy), Ctx(Ctx) {}
```

Comment:
nit (unrelated to this patch): does VPTypeAnalysis really need `Ctx`?

Reply:
Is there any current use of `Ctx`? Sure, if this is indeed dead, should be eliminated separately.

Reply:
The current uses are to create `i1` & `void` types:

https://github.com/llvm/llvm-project/blob/main/llvm/lib/Transforms/Vectorize/VPlanAnalysis.cpp#L63
https://github.com/llvm/llvm-project/blob/main/llvm/lib/Transforms/Vectorize/VPlanAnalysis.cpp#L163
https://github.com/llvm/llvm-project/blob/main/llvm/lib/Transforms/Vectorize/VPlanAnalysis.cpp#L189